File: aiAzure/python/testAzureAssistant.py
from dotenv import load_dotenv
import os
import time
from openai import AzureOpenAI
import sys

load_dotenv()

def initialize_client(): 
    return AzureOpenAI(
        api_key= os.getenv('API_KEY'),
        api_version="2024-05-01-preview",
        azure_endpoint = "https://craftersopenai.openai.azure.com/"
        )

def create_thread(client): 
    return  client.beta.threads.create()

def add_message_to_thread(client, thread_id, message): 
    client.beta.threads.messages.create(
        thread_id=thread_id,
        role="user",
        content=message # Replace this with your prompt
    ) 

def create_run(client, thread_id, assistant_id):
    return client.beta.threads.runs.create(
        thread_id=thread_id,
        assistant_id=assistant_id
    )

def retrieve_run(client, thread_id, run_id):
    return client.beta.threads.runs.retrieve(
                thread_id=thread_id,
                run_id=run_id
            )      

def retrieve_response(client, thread_id):
      thread_messages = client.beta.threads.messages.list(thread_id)
      return thread_messages.data[0].content[0].text.value

def main():
    client = initialize_client()

    thread = create_thread(client)

    message = sys.argv[1]
    add_message_to_thread(client, thread.id, message)

    run = create_run(client, thread.id, "asst_bhRd5zAq1b4MgY7oUIlqYucm")

    while True:
        run = retrieve_run(client, thread.id, run.id)        
        status = run.status
        
        if status == 'completed':
            break
        time.sleep(1)  # Adjust polling interval as needed

    print(f"Response: {retrieve_response(client, thread.id)}")
    
if __name__ == "__main__":
    main()

File: aiAzure/node/testAzureAssistant.js
import { AzureKeyCredential } from '@azure/openai';
import { AssistantsClient } from '@azure/openai-assistants';
import readline from "readline";
import "dotenv/config.js";

const endpoint = process.env.AZURE_OPENAI_ENDPOINT || 'https://craftersopenai.openai.azure.com/';
const azureApiKey = process.env.API_KEY;
const assistantId = 'asst_TlosK9CVyvbV03MVYhV3M1iw'; // ENTER YOUR ASSISTANT ID

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout
});

async function sendMessage(message) {
  const assistantsClient = new AssistantsClient(endpoint, new AzureKeyCredential(azureApiKey));
  const assistantThread = await assistantsClient.createThread();
  await assistantsClient.createMessage(assistantThread.id, "user", message);
  let runResponse = await assistantsClient.createRun(assistantThread.id, {
    assistantId: assistantId
  });  
 do {
    await new Promise((resolve) => setTimeout(resolve, 800));
    runResponse = await assistantsClient.getRun(assistantThread.id, runResponse.id);
  } while (runResponse.status === "queued" || runResponse.status === "in_progress") 
    const runMessages = await assistantsClient.listMessages(assistantThread.id);
    const {data} = runMessages;
    const {content} = data[0];
    console.log(`Response: ${content[0].text.value}`);  
}

function main() {
  rl.question('Enter your message: ', async (message) => {
    await sendMessage(message);
    main(); // call main again to prompt for next message
  });
}

main();